annotations/builder/src/main/java/io/sundr/builder/internal/utils/BuilderUtils.java

```diff
@@ -423,7 +425,7 @@ public static String fullyQualifiedNameDiff(TypeRef typeRef, TypeDef originType)
         String currentPackage = originType != null ? originType.getPackageName() : null;
 
         if (typeRef instanceof ClassRef) {
-            ClassRef classRef = (ClassRef)  TypeAs.combine(UNWRAP_COLLECTION_OF, UNWRAP_ARRAY_OF).apply(typeRef);
+            ClassRef classRef = (ClassRef)  TypeAs.combine(UNWRAP_COLLECTION_OF, UNWRAP_ARRAY_OF, UNWRAP_OPTIONAL_OF).apply(typeRef);
```

Comment:
@jimmidyson: You can't just cast a TypeRef to ClassRef without checking. I know that the codebase is allover this kind of cast, but this is only the case, when we are dealing with types recognized as buildables. So we now that they are not primitves, they are not generics, or null types.